src/ruby/ext/symengine/ruby_symbol.h

```diff
@@ -0,0 +1,9 @@
+#include "ruby.h"
+#include "ruby_basic.h"
+#include "cwrapper.h"
+
+void csymbol_free(void *ptr);
```

Comment:
When we make an object of a certain class of Ruby in the Ruby wrapper code, we use a method `Data_Make_Struct` as used [here](https://github.com/sympy/symengine/pull/480/files#diff-4c1a05fa0d0faea4de4f1b04b88cce71R47). It allocates the space for the type, wraps the c pointer for it with the Ruby class we define. It also takes the function to free the wrapped c pointer, to be used when Garbage Collector is called on the formed object.
More can be found [here](http://clalance.blogspot.in/2013/11/writing-ruby-extensions-in-c-part-13.html)